pkg/machine/docker.go

```diff
@@ -30,9 +31,9 @@ func (q *Docker) Create(ctx context.Context) error {
 	cmd := fmt.Sprintf("%s run %s --entrypoint /bin/sh -d -t --name %s %s", processName, strings.Join(q.machineConfig.Args, " "), q.machineConfig.ID, q.machineConfig.Image)
 	out, err := utils.SH(cmd)
 	if err != nil {
-		return fmt.Errorf("failed creating container: %w - cmd: %s, out: %s", err, cmd, out)
+		return ctx, fmt.Errorf("failed creating container: %w - cmd: %s, out: %s", err, cmd, out)
 	}
-	return nil
+	return ctx, nil
```

Comment:
How is this ctx different from the one given as an argument? feels to me like instead of passing a copy we should be passing a pointer

Reply:
In this case, there is no difference. I simply return a context so that it satisfies the interface. The new context is different only in the case of qemu. There, we can't use the one from the argument because that my not have a cancel function which we need in order to control when the context is "Done" (when the process exits).

Reply:
gotcha, thanks for clarifying